src/js/importer.js

```diff
@@ -0,0 +1,374 @@
+                                    verses.push(v);
+                                }
+
+                            //if it doesn't exist or it's not the same chapter as the current chunk, add 100 verses so we make sure to get everything being imported.
+                            } else {
+                                var v = parseInt(chunk.firstvs), max = v + 100;
+                                for(v;v<max;v++){
+                                    verses.push(v);
+                                }
+                            }
+                            var chk = {
+                                filename: chunk.firstvs,
+                                chapter: chunk.chp,
+                                verses: verses
+                            };
+                            chunks.push(chk);
+                        }
+                        resolve(chunks);
+                    } else {
+                        reject(err);
+                    }
+                });
+            });
+        },
+
+        /*getVerseChunkFileNames: function(projdata){
+            return new Promise(function(resolve,reject){
+                var frames = pm.getSourceFrames(projdata.source_translations[projdata.currentsource]);
+                var parsedFrames = [],
+                    chapters = {};
+
+                for(var i = 0;i < frames.length;i++){
+                    var sourcedata = frames[i].chunk,
+                        verses = [],
+                        startstr = '<verse number="',
+                        endstr = '" style="v" />',
+                        test = new RegExp(startstr);;
+
+                    while (test.test(sourcedata)) {
+                        var versestr = sourcedata.substring(sourcedata.search(startstr) + 15, sourcedata.search(endstr));
+                        if (versestr.indexOf("-") < 0) {
+                            verses.push(parseInt(versestr));
+                        } else {
+                            var firstnum = parseInt(versestr.substring(0, versestr.search("-")));
+                            var lastnum = parseInt(versestr.substring(versestr.search("-") + 1));
+                            for (var j = firstnum; j <= lastnum; j++) {
+                                verses.push(j);
+                            }
+                        }
+                        sourcedata = sourcedata.replace(startstr, " \\v");
+                        sourcedata = sourcedata.replace(endstr, " ");
+                    }
+
+                    var chuckFile = String("00" + verses[0]).slice(-2);
+
+                    parsedFrames.push({filename:chuckFile,chapter:frames[i].chapter,verses:verses});
+                }
+                resolve(parsedFrames);
+            });
+        },*/
+
+        importSingleUSFMFile: function (usfmFile, projdata, user) {
+            let self = this;
+            return pm.loadTargetTranslation(projdata).then(function(curProj){
```

Comment:
at this point we should only have the usfm file, the project, the target language, and the user.
We will create a new target translation and return that. We are not opening an existing target translation.